view/perspective/app.js

```diff
@@ -115,34 +119,61 @@ function handleError(err) {
 function handleEvent(eventData, eventTypeName, trackEndToEndTime) {
   trackEndToEndTime && trackEndToEndTime(Date.now());
 
-  const j = JSON.parse(eventData);
+  const obj = JSON.parse(eventData)[eventTypeName];
   if (DEBUG_REALTIME) {
     console.log({ // eslint-disable-line no-console
       handleEventTimestamp: new Date(),
-      eventData: j,
+      eventData: obj,
     });
   }
 
-  eventsQueue.enqueueEvent(eventTypeName, j[eventTypeName]);
+  // intercept events to support v1 lenses, and track timeouts for auto-reload.
+  interceptEvent(eventTypeName, obj);
+
+  eventsQueue.enqueueEvent(eventTypeName, obj);
   if (_realtimeEventThrottleMilliseconds === ZERO) {
     eventsQueue.createAndDispatchLensEvent(eventsQueue.queue, LENS_DIV);
     eventsQueue.queue.length = ZERO;
   }
+} // handleEvent
 
-  if (eventTypeName === eventsQueue.eventType.INTRNL_SMPL_ADD) {
-    const sample = j[eventTypeName];
-    updateTimeoutValues(sample.aspect.timeout);
-  } else if (eventTypeName === eventsQueue.eventType.INTRNL_SMPL_UPD) {
-    const newSample = j[eventTypeName].new;
-    updateTimeoutValues(newSample.aspect.timeout);
-  } else if (eventTypeName === eventsQueue.eventType.INTRNL_SMPL_DEL) {
-    const sample = j[eventTypeName];
-    updateDeletedTimeoutValues(sample.aspect.timeout);
-  } else if (eventTypeName === eventsQueue.eventType.INTRNL_SMPL_NC) {
-    const sample = j[eventTypeName];
-    updateTimeoutValues(sample.aspect.timeout);
+/**
+ * Intercept events. Attach aspects to support v1 lenses, and track timeouts
+ * for auto-reload.
+ *
+ * @param  {String} eventTypeName - Event type
+ * @param  {Object} eventData - object that will be sent to the lens
+ */
+function interceptEvent(eventTypeName, eventData) {
+  if (lensEventApiVersion < 2) {
+    if (eventTypeName === eventsQueue.eventType.INTRNL_SMPL_ADD) {
+      eventData.aspect = getTrackedAspect(eventData);
+    } else if (eventTypeName === eventsQueue.eventType.INTRNL_SMPL_DEL) {
+      eventData.aspect = getTrackedAspect(eventData);
+    } else if (eventTypeName === eventsQueue.eventType.INTRNL_SMPL_UPD) {
+      eventData.new.aspect = getTrackedAspect(eventData.new);
+      eventData.old.aspect = getTrackedAspect(eventData.new);
```

Comment:
can be the case here as ```eventData.old.aspect = eventData.new.aspect``` ?

Reply:
Yeah, we're only tracking the latest aspect, so we set both old and new to use that one. It's technically not completely the same as it would have been before, but I doubt there's any lens that relies on that behavior, and tracking the previous version of the aspect would significantly complicate things.

Reply:
👍🏻